### ISSUE TITLE
Some commands don't work

### ISSUE THREAD

**Issue by ajh123** [Team-Inceptus/Novaconomy#117]
### Bug Type

Console Errors, Other Unexpected Behavior

### Minecraft Version

1.20.1 (any MC version that can run Novaconomy 1.8.1-SNAPSHOT)

### Description

Some commands are not working properly. Some say "invalid arguments" when they are correct or some say "an error occurred while executing this command".

### Steps to Reproduce

1. Run `/novaconomy addbal Samland_Pound samuelh2005 100` (replace "samuelh2005" with your name and replace "Samland_Pound" with a valid economy on your server)
2. Run `/business create StuffInc bookshelf`

### Expected Behavior

Both commands work correctly and produce no errors.

### Actual Behavior

First command fails and error produced:
```
[15:34:55] [Server thread/ERROR] [STDERR/]: java.lang.NullPointerException: Cannot read field "moneyAdded" because "this.stats" is null
[15:34:55] [Server thread/ERROR] [STDERR/]:     at us.teaminceptus.novaconomy.api.player.NovaPlayer.add(NovaPlayer.java:281)
[15:34:55] [Server thread/ERROR] [STDERR/]:     at us.teaminceptus.novaconomy.abstraction.CommandWrapper.addBalance(CommandWrapper.java:421)
[15:34:55] [Server thread/ERROR] [STDERR/]:     at us.teaminceptus.novaconomy.CommandWrapperV2$EconomyCommands.addBalance(CommandWrapperV2.java:550)
```
The second one says it worked in the chat but in the console it produces
```
[16:11:08] [Server thread/ERROR] [Novaconomy/]: NotSerializableException
[16:11:08] [Server thread/ERROR] [Novaconomy/]: -----------
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.settings.Settings$Business
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1197)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1943)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.writeObject(HashMap.java:1497)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.reflect.Method.invoke(Method.java:568)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1201)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1526)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1448)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1191)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.writeDB(Business.java:987)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.saveBusiness(Business.java:907)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business$Builder.build(Business.java:2064)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.abstraction.CommandWrapper.createBusiness(CommandWrapper.java:783)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.CommandWrapperV2$BusinessCommands.businessCreate(CommandWrapperV2.java:339)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:757)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodHandlesCallerFactory$1.call(MethodHandlesCallerFactory.java:58)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodCaller.lambda$bindTo$0(MethodCaller.java:52)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.execute(BaseCommandDispatcher.java:111)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.searchCategory(BaseCommandDispatcher.java:84)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.eval(BaseCommandDispatcher.java:67)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandHandler.dispatch(BaseCommandHandler.java:540)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.bukkit.core.BukkitCommandExecutor.onCommand(BukkitCommandExecutor.java:32)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServerMixin.java:875)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:50)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: MC-BOOTSTRAP/brigadier@1.1.8/com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.commands.Commands.m_242674_(CommandsMixin.java:257)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_246958_(ServerPlayNetHandlerMixin.java:2767)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_244885_(ServerPlayNetHandlerMixin.java:1221)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18693_(BlockableEventLoop.java:67)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.TickTask.run(TickTask.java:18)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:770)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:161)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServerMixin.java:753)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServerMixin.java:747)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServerMixin.java:733)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServerMixin.java:2074)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServerMixin.java:251)
[16:11:08] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.Thread.run(Thread.java:833)
```

### Additional Info

I am using Arclight but the error originates in Novaconomy, so the error should happen on other Bukkit / Spigot / Paper implementations. Also I'm not sure on how to fix this one.

**Comment by GamerCoder215** [comment]
Combined issue with the outdated command framework (Lamp was updated recently) and some other command wrapper errors.

**Comment by GamerCoder215** [comment]
Draft commit in e5d9ec7. Not tested.

**Comment by ajh123** [comment]
The first command works as expected but the second one still produces the same error. At least were closer now :)

**Comment by GamerCoder215** [comment]
> The first command works as expected but the second one still produces the same error. At least were closer now :)

Serializale exceptions are pretty unreversable; it's using an Object reference that isn't Serializale and changed with new versions. Recovery may be possible but unlikely on a SQL Database. I'll do more testing, so don't do anything yet.

**Comment by ajh123** [comment]
My businesses table is empty so there is not data in it. I can delete it if you need me to.

**Comment by GamerCoder215** [comment]
Try deleting it; bStats says that your plugin is the only one in production with SQL, so there probably won't be a large effect. Just in case, try it and see.

**Comment by ajh123** [comment]
When command 2 is ran the error still exists also chat says "Please provide valid arguments". Seems like there are problems with validation.

**Comment by GamerCoder215** [comment]
> When command 2 is ran the error still exists also chat says "Please provide valid arguments". There are problems with validation.

I looked at the code and because the command deals with business **creation** instead of changing regimes, no production servers should be affected, along with the reasons I stated above. I updated Lamp to v3.1.7, and the commands seem to be working a lot better.

**Comment by ajh123** [comment]
I've updated to the newest version but command 2 still creates and error! 
```
[11:59:14] [Server thread/ERROR] [Novaconomy/]: NotSerializableException
[11:59:14] [Server thread/ERROR] [Novaconomy/]: -----------
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.settings.Settings$Business$$Lambda$17656/0x0000000802fbec68
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1197)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1582)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1539)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1448)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1191)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1943)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.writeObject(HashMap.java:1497)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.reflect.Method.invoke(Method.java:568)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1201)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1526)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1448)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1191)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.writeDB(Business.java:987)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.saveBusiness(Business.java:907)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business$Builder.build(Business.java:2064)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.abstraction.CommandWrapper.createBusiness(CommandWrapper.java:784)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.CommandWrapperV2$BusinessCommands.businessCreate(CommandWrapperV2.java:375)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:757)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodHandlesCallerFactory$1.call(MethodHandlesCallerFactory.java:58)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodCaller.lambda$bindTo$0(MethodCaller.java:52)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.execute(BaseCommandDispatcher.java:113)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.searchCategory(BaseCommandDispatcher.java:86)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.eval(BaseCommandDispatcher.java:69)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandHandler.dispatch(BaseCommandHandler.java:540)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.bukkit.core.BukkitCommandExecutor.onCommand(BukkitCommandExecutor.java:35)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServerMixin.java:875)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:50)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: MC-BOOTSTRAP/brigadier@1.1.8/com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.commands.Commands.m_242674_(CommandsMixin.java:257)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_246958_(ServerPlayNetHandlerMixin.java:2767)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_244885_(ServerPlayNetHandlerMixin.java:1221)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18693_(BlockableEventLoop.java:67)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.TickTask.run(TickTask.java:18)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:770)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:161)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServerMixin.java:753)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServerMixin.java:747)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServerMixin.java:733)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServerMixin.java:2074)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServerMixin.java:251)
[11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by GamerCoder215** [comment]
> @GamerCoder215
> 
> I've updated to the newest version but command 2 still creates and error!
> 
> ```
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: NotSerializableException
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: -----------
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.settings.Settings$Business$$Lambda$17656/0x0000000802fbec68
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1197)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1582)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1539)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1448)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1191)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1943)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.HashMap.writeObject(HashMap.java:1497)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.reflect.Method.invoke(Method.java:568)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1201)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1526)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1448)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1191)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.writeDB(Business.java:987)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business.saveBusiness(Business.java:907)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.api.business.Business$Builder.build(Business.java:2064)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.abstraction.CommandWrapper.createBusiness(CommandWrapper.java:784)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.CommandWrapperV2$BusinessCommands.businessCreate(CommandWrapperV2.java:375)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:757)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodHandlesCallerFactory$1.call(MethodHandlesCallerFactory.java:58)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.reflect.MethodCaller.lambda$bindTo$0(MethodCaller.java:52)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.execute(BaseCommandDispatcher.java:113)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.searchCategory(BaseCommandDispatcher.java:86)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandDispatcher.eval(BaseCommandDispatcher.java:69)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.core.BaseCommandHandler.dispatch(BaseCommandHandler.java:540)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: us.teaminceptus.novaconomy.shaded.lamp.bukkit.core.BukkitCommandExecutor.onCommand(BukkitCommandExecutor.java:35)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.CraftServer.dispatchCommand(CraftServerMixin.java:875)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/arclight@1.20.1-1.0.0-053d747/org.bukkit.craftbukkit.v1_20_R1.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:50)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: MC-BOOTSTRAP/brigadier@1.1.8/com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:264)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.commands.Commands.m_242674_(CommandsMixin.java:257)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_246958_(ServerPlayNetHandlerMixin.java:2767)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.network.ServerGamePacketListenerImpl.m_244885_(ServerPlayNetHandlerMixin.java:1221)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18693_(BlockableEventLoop.java:67)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.TickTask.run(TickTask.java:18)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:770)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_6367_(MinecraftServerMixin.java:161)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServerMixin.java:753)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServerMixin.java:747)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServerMixin.java:733)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServerMixin.java:2074)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServerMixin.java:251)
> [11:59:14] [Server thread/ERROR] [Novaconomy/]: java.base/java.lang.Thread.run(Thread.java:833)
> ```

Weird. My SQL test came out fine. Might have some differences in Serializability for Functional Interfaces per JDK distribution, so I just got rid of it entirely.

**Comment by GamerCoder215** [comment]
(Should be) Fixed and tested in 16ed7c7. I caught a few more SQL and SQL Conversion Errors that showed up, so we should be good. Feel free to test on your own.
